### ISSUE TITLE
Issue running the automl object detection fridgeitems

### ISSUE THREAD

**Issue by sruthivijay** [Azure/azureml-examples#1382]
## Automl Object detection for fridge items:
The example notebook for automl object detection of fridge items job fails while running without producing clear errors logs.
example: driver logs for the same has been attached . It is difficult to understand the issue from the drivers log.notebook has been ran as it is.
I am tryin to use the code for custom training, but the example itself fails. Can you help figure out? with the previous version code it runs without any issue.
description:

## Additional context

-{'device': device(type='cuda', index=0), 'primary_metric': 'mean_average_precision', 'number_of_epochs': 30, 'training_batch_size': 12, 'validation_batch_size': 16, 'learning_rate': 0.004938355433218799, 'early_stopping': True, 'early_stopping_delay': 5, 'early_stopping_patience': 5, 'grad_accumulation_step': 1, 'grad_clip_type': 'value', 'optimizer': 'sgd', 'momentum': 0.9, 'weight_decay': 0.0001, 'nesterov': True, 'beta1': 0.9, 'beta2': 0.999, 'amsgrad': False, 'lr_scheduler': 'warmup_cosine', 'step_lr_gamma': 0.5, 'step_lr_step_size': 5, 'warmup_cosine_lr_cycles': 0.45, 'warmup_cosine_lr_warmup_epochs': 2, 'evaluation_frequency': 1, 'validation_size': 0.2, 'enable_onnx_normalization': False, 'ignore_data_errors': True, 'log_scoring_file_info': False, 'model_name': 'yolov5', 'num_workers': 8, 'output_scoring': False, 'validate_scoring': False, 'save_mlflow': True, 'validation_metric_type': 'voc', 'validation_iou_threshold': 0.5, 'tile_overlap_ratio': 0.25, 'tile_predictions_nms_thresh': 0.25, 'distributed': True, 'MASTER_ADDR': '127.0.0.1', 'MASTER_PORT': '29500', 'seed': 1, 'checkpoint_frequency': None, 'checkpoint_run_id': None, 'checkpoint_dataset_id': None, 'checkpoint_filename': None, 'layers_to_freeze': None, 'tile_grid_size': None, 'img_size': 640, 'model_size': 'small', 'multi_scale': False, 'box_score_thresh': 0.1, 'nms_iou_thresh': 0.5, 'subscription_id': '5f854be0-ae65-4afa-9f92-8aeba71c0608', 'resource_group': 'usedmlt00wrsg01', 'workspace_name': 'USEDMLT00WMLS01', 'compute_target': 'MLT00W-ML-CC', 'iterations': 1, 'task_type': 'image-object-detection', 'n_cross_validations': None, 'preprocess': False, 'max_cores_per_iteration': -1, 'max_concurrent_iterations': 1, 'enforce_time_on_windows': False, 'experiment_timeout_minutes': 60, 'experiment_exit_score': 'NaN', 'auto_blacklist': False, 'blacklist_samples_reached': False, 'exclude_nan_labels': False, 'verbosity': 20, 'model_explainability': False, 'enable_onnx_compatible_models': False, 'enable_feature_sweeping': False, 'send_telemetry': True, 'enable_early_stopping': True, 'early_stopping_n_iters': 20, 'ensemble_iterations': 1, 'enable_tf': False, 'enable_subsampling': False, 'metric_operation': 'maximize', 'enable_streaming': False, 'force_streaming': False, 'enable_dnn': True, 'featurization': 'off', 'weight_column_name': None, 'many_models': False, 'track_child_runs': True, 'enable_ensembling': False, 'enable_stack_ensembling': False, 'blacklist_algos': ['TensorFlowDNN', 'TensorFlowLinearRegressor']}
Settings not logged (might contain pii): 
 {'fl_gamma', 'enable_cache', 'use_incremental_learning_override', 'scale', 'data_folder', 'exit_score', 'is_timeseries', 'giou', 'anchor_t', 'grain_column_names', 'model', 'degrees', 'enable_distributed_dnn_training', 'automl_many_models_scenario', 'miro_flight', 'enable_distributed_featurization', 'label_column_name', 'gs', 'enable_mltable_quick_profile', 'time_column_name', 'start_auxiliary_runs_before_parent_complete', 'enable_run_restructure', 'ensemble_download_models_timeout_sec', 'obj', 'cls_pw', 'path', 'enable_batch_run', 'hyperdrive_config', 'IsTextDNNTask', 'test_include_predictions_only', 'translate', 'labels_file_root', 'IsImageTask', 'many_models_process_count_per_node', 'cls', 'stack_meta_learner_train_percentage', 'shear', 'split_ratio', 'enable_distributed_dnn_training_ort_ds', 'obj_pw', 'is_gpu_tmp'}
Type: Unclassified
Class: ValueError
Message: [Hidden as it may contain PII]
Traceback:   File "utils.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "runner.py", line 83, in run
    train_ds, validation_ds = utils.get_tabular_dataset(settings=settings, mltable_json=mltable_data_json)
  File "utils.py", line 885, in get_tabular_dataset
    mltable_json, ws, MLTableDataLabel.TrainData)
  File "utils.py", line 827, in get_dataset_from_mltable
    return get_dataset_from_mltable_data_json(workspace, mltable_data, data_label)
  File "dataset_utilities.py", line 139, in get_dataset_from_mltable_data_json
    dataset = AbstractDataset._load(dataset_uri, ws)
  File "_loggerfactory.py", line 132, in wrapper
    return func(*args, **kwargs)
  File "abstract_dataset.py", line 730, in _load
    local_path = _download_mltable_yaml(path)
  File "_mltable_helper.py", line 36, in _download_mltable_yaml
    raise ValueError(e.args[0])

ExceptionTarget: Unspecified
Marking Run clever_parrot_77lmzvtr4q_HD_0 as Failed.
Type: SystemError
Class: ClientException
Message: ClientException:
	Message: Encountered an internal AutoML Image error. Error Message/Code: {error_details}. Traceback:   File "utils.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "runner.py", line 83, in run
    train_ds, validation_ds = utils.get_tabular_dataset(settings=settings, mltable_json=mltable_data_json)
  File "utils.py", line 885, in get_tabular_dataset
    mltable_json, ws, MLTableDataLabel.TrainData)
  File "utils.py", line 827, in get_dataset_from_mltable
    return get_dataset_from_mltable_data_json(workspace, mltable_data, data_label)
  File "dataset_utilities.py", line 139, in get_dataset_from_mltable_data_json
    dataset = AbstractDataset._load(dataset_uri, ws)
  File "_loggerfactory.py", line 132, in wrapper
    return func(*args, **kwargs)
  File "abstract_dataset.py", line 730, in _load
    local_path = _download_mltable_yaml(path)
  File "_mltable_helper.py", line 36, in _download_mltable_yaml
    raise ValueError(e.args[0])
. Additional information: [Hidden as it may contain PII].
	InnerException: ValueError
	ErrorResponse 
{
    "error": {
        "code": "SystemError",
        "message": "Encountered an internal AutoML Image error. Error Message/Code: {error_details}. Traceback:   File \"utils.py\", line 311, in wrapper\n    return func(*args, **kwargs)\n  File \"runner.py\", line 83, in run\n    train_ds, validation_ds = utils.get_tabular_dataset(settings=settings, mltable_json=mltable_data_json)\n  File \"utils.py\", line 885, in get_tabular_dataset\n    mltable_json, ws, MLTableDataLabel.TrainData)\n  File \"utils.py\", line 827, in get_dataset_from_mltable\n    return get_dataset_from_mltable_data_json(workspace, mltable_data, data_label)\n  File \"dataset_utilities.py\", line 139, in get_dataset_from_mltable_data_json\n    dataset = AbstractDataset._load(dataset_uri, ws)\n  File \"_loggerfactory.py\", line 132, in wrapper\n    return func(*args, **kwargs)\n  File \"abstract_dataset.py\", line 730, in _load\n    local_path = _download_mltable_yaml(path)\n  File \"_mltable_helper.py\", line 36, in _download_mltable_yaml\n    raise ValueError(e.args[0])\n. Additional information: [Hidden as it may contain PII].",
        "details_uri": "https://aka.ms/automltroubleshoot",
        "inner_error": {
            "code": "ClientError",
            "inner_error": {
                "code": "AutoMLInternal",
                "inner_error": {
                    "code": "AutoMLVisionInternal"
                }
            }
        }
    }
}
Traceback:   File "utils.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "runner.py", line 83, in run
    train_ds, validation_ds = utils.get_tabular_dataset(settings=settings, mltable_json=mltable_data_json)
  File "utils.py", line 885, in get_tabular_dataset
    mltable_json, ws, MLTableDataLabel.TrainData)
  File "utils.py", line 827, in get_dataset_from_mltable
    return get_dataset_from_mltable_data_json(workspace, mltable_data, data_label)
  File "dataset_utilities.py", line 139, in get_dataset_from_mltable_data_json
    dataset = AbstractDataset._load(dataset_uri, ws)
  File "_loggerfactory.py", line 132, in wrapper
    return func(*args, **kwargs)
  File "abstract_dataset.py", line 730, in _load
    local_path = _download_mltable_yaml(path)
  File "_mltable_helper.py", line 36, in _download_mltable_yaml
    raise ValueError(e.args[0])

ExceptionTarget: Unspecified
Encountered an exception of type: <class 'azureml.automl.core.shared.exceptions.ClientException'>, interpreted as error code AutoMLVisionInternal.
Encountered an exception of type: <class 'azureml.automl.core.shared.exceptions.ClientException'>, interpreted as error code SystemError.ClientError.AutoMLInternal.AutoMLVisionInternal.
Type: Unclassified
Class: ValueError
Message: [Hidden as it may contain PII]
Traceback:   File "utils.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "runner.py", line 72, in run
    yolo_runner.run(automl_settings, mltable_data_json)
  File "utils.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "runner.py", line 83, in run
    train_ds, validation_ds = utils.get_tabular_dataset(settings=settings, mltable_json=mltable_data_json)
  File "utils.py", line 885, in get_tabular_dataset
    mltable_json, ws, MLTableDataLabel.TrainData)
  File "utils.py", line 827, in get_dataset_from_mltable
    return get_dataset_from_mltable_data_json(workspace, mltable_data, data_label)
  File "dataset_utilities.py", line 139, in get_dataset_from_mltable_data_json
    dataset = AbstractDataset._load(dataset_uri, ws)
  File "_loggerfactory.py", line 132, in wrapper
    return func(*args, **kwargs)
  File "abstract_dataset.py", line 730, in _load
    local_path = _download_mltable_yaml(path)
  File "_mltable_helper.py", line 36, in _download_mltable_yaml
    raise ValueError(e.args[0])

ExceptionTarget: Unspecified
Marking Run clever_parrot_77lmzvtr4q_HD_0 as Failed.
Type: SystemError
Class: ClientException
Message: ClientException:
	Message: Encountered an internal AutoML Image error. Error Message/Code: {error_details}. Traceback:   File "utils.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "runner.py", line 72, in run
    yolo_runner.run(automl_settings, mltable_data_json)
  File "utils.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "runner.py", line 83, in run
    train_ds, validation_ds = utils.get_tabular_dataset(settings=settings, mltable_json=mltable_data_json)
  File "utils.py", line 885, in get_tabular_dataset
    mltable_json, ws, MLTableDataLabel.TrainData)
  File "utils.py", line 827, in get_dataset_from_mltable
    return get_dataset_from_mltable_data_json(workspace, mltable_data, data_label)
  File "dataset_utilities.py", line 139, in get_dataset_from_mltable_data_json
    dataset = AbstractDataset._load(dataset_uri, ws)
  File "_loggerfactory.py", line 132, in wrapper
    return func(*args, **kwargs)
  File "abstract_dataset.py", line 730, in _load
    local_path = _download_mltable_yaml(path)
  File "_mltable_helper.py", line 36, in _download_mltable_yaml
    raise ValueError(e.args[0])
. Additional information: [Hidden as it may contain PII].
	InnerException: ValueError
	ErrorResponse 
{
    "error": {
        "code": "SystemError",
        "message": "Encountered an internal AutoML Image error. Error Message/Code: {error_details}. Traceback:   File \"utils.py\", line 311, in wrapper\n    return func(*args, **kwargs)\n  File \"runner.py\", line 72, in run\n    yolo_runner.run(automl_settings, mltable_data_json)\n  File \"utils.py\", line 311, in wrapper\n    return func(*args, **kwargs)\n  File \"runner.py\", line 83, in run\n    train_ds, validation_ds = utils.get_tabular_dataset(settings=settings, mltable_json=mltable_data_json)\n  File \"utils.py\", line 885, in get_tabular_dataset\n    mltable_json, ws, MLTableDataLabel.TrainData)\n  File \"utils.py\", line 827, in get_dataset_from_mltable\n    return get_dataset_from_mltable_data_json(workspace, mltable_data, data_label)\n  File \"dataset_utilities.py\", line 139, in get_dataset_from_mltable_data_json\n    dataset = AbstractDataset._load(dataset_uri, ws)\n  File \"_loggerfactory.py\", line 132, in wrapper\n    return func(*args, **kwargs)\n  File \"abstract_dataset.py\", line 730, in _load\n    local_path = _download_mltable_yaml(path)\n  File \"_mltable_helper.py\", line 36, in _download_mltable_yaml\n    raise ValueError(e.args[0])\n. Additional information: [Hidden as it may contain PII].",
        "details_uri": "https://aka.ms/automltroubleshoot",
        "inner_error": {
            "code": "ClientError",
            "inner_error": {
                "code": "AutoMLInternal",
                "inner_error": {
                    "code": "AutoMLVisionInternal"
                }
            }
        }
    }
}
Traceback:   File "utils.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "runner.py", line 72, in run
    yolo_runner.run(automl_settings, mltable_data_json)
  File "utils.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "runner.py", line 83, in run
    train_ds, validation_ds = utils.get_tabular_dataset(settings=settings, mltable_json=mltable_data_json)
  File "utils.py", line 885, in get_tabular_dataset
    mltable_json, ws, MLTableDataLabel.TrainData)
  File "utils.py", line 827, in get_dataset_from_mltable
    return get_dataset_from_mltable_data_json(workspace, mltable_data, data_label)
  File "dataset_utilities.py", line 139, in get_dataset_from_mltable_data_json
    dataset = AbstractDataset._load(dataset_uri, ws)
  File "_loggerfactory.py", line 132, in wrapper
    return func(*args, **kwargs)
  File "abstract_dataset.py", line 730, in _load
    local_path = _download_mltable_yaml(path)
  File "_mltable_helper.py", line 36, in _download_mltable_yaml
    raise ValueError(e.args[0])

ExceptionTarget: Unspecified
Encountered an exception of type: <class 'azureml.automl.core.shared.exceptions.ClientException'>, interpreted as error code AutoMLVisionInternal.
Encountered an exception of type: <class 'azureml.automl.core.shared.exceptions.ClientException'>, interpreted as error code SystemError.ClientError.AutoMLInternal.AutoMLVisionInternal.

**Comment by jsamples121g** [comment]
I am having the same issue running the notebook.  The basis of the error is 
Encountered an internal AutoML Image error. Error Message/Code: Failed to load mltable to dataset with exception: 
Error Code: Validation
Validation Error Code: Invalid MLTable
Validation Target: MLTableToDataflow
Error Message: Failed to convert a MLTable to dataflow
uri path is not a valid datastore uri path

**Comment by bvillab** [comment]
Same here. I have been trying to deploy a new ML Model with the new paremeters and new library and it is not working. The documentation is not helpful at all and the example here in github is not working too.

**Comment by sruthivijay** [comment]
@jsamples121g @bvillab 
Hey guys,
    a quick fix for the above suggested by microsoft support team. I don't think the URI reference works. So it's better to create tabular dataset like mentioned in the previous version and refer them like this
my_training_data_input = Input(type=AssetTypes.MLTABLE, path="azureml:odFridgeObjectsTrainingDataset:1")
my_validation_data_input = Input(type=AssetTypes.MLTABLE, path="azureml:odFridgeObjectsValidationDataset:1")

1 represents the version
this worked for me

**Comment by raduk** [comment]
These errors come from 2 sources:

1.  The MLTable file is missing. For the notebooks in this repository, this file is checked in along with the notebook, so if the notebook is copied in another location, that file is not found. This is not ideal, and we'll modify the notebooks to create the MLTable on the fly.
2. The MLTable file is invalid, for instance the column containing the images is not marked as stream.

The below image explains these 2 issues. We also worked on modifying the error messages to be more explicit. 

![image](https://user-images.githubusercontent.com/3809388/203128533-1029dcf3-045c-4475-aa46-d389dc126f48.png)
